edx_proctoring/utils.py

```diff
@@ -2,12 +2,18 @@
 Helpers for the HTTP APIs
 """
 
+import base64
 import hashlib
 import hmac
 import logging
 from datetime import datetime, timedelta
 
 import pytz
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.ciphers import Cipher
+from cryptography.hazmat.primitives.ciphers.algorithms import AES
+from cryptography.hazmat.primitives.ciphers.modes import CBC
+from cryptography.hazmat.primitives.padding import PKCS7
```

Comment:
Based on https://cryptography.io/en/latest/#layout it sounds like the choice to import from hazmat was made in the past by folks who really deeply understand the cryptography at play, something I'm not sure we understand at this point. I wanted to flag this because the same section refers to a closer-to-the-surface and "more reasonable defaults" possible imports offered from the same library, which I'm wondering if we explored using.

If this specific way of encrypting/decrypting is required by the vendor, then we don't really have a choice here and should probably just resolve this comment without taking action. But I'd like to confirm that's the case, if possible.

Reply:
Hmm, I had not explored other import options, but saw that `hazmat` is used for both IDV and in `UsernameCipher`: https://github.com/edx/edx-platform/blob/3152c14192dcccac79b7b7a76eba91b380eba78a/lms/djangoapps/discussion/notification_prefs/views.py#L34. 

For this PR, we're not tied to a specific way of encrypting/decrypting because we (edX) will be the only user/consumer of this key.

Reply:
Could we write a follow-up ticket to explore refactoring this to use more tried-and-true encryption strategies? I imagine we're copying and pasting code here to avoid expanding scope into the whole rabbit hole of cryptographic algorithms, but maybe it makes sense to spend some time seeing whether we can use this library in a safer way?